hw2/02-reverse.js

```diff
@@ -1,3 +1,30 @@
-/** Exercise 02 - Reverse **/
+/** Exercise 02 - Reverse * */
+const input = document.querySelector('#input')
+const reverse = document.querySelector('#reverse')
+const result = document.querySelector('#result')
 
-// Add your code here
+const setError = (msg) => {
+    result.style.color = '#990505'
+    result.textContent = `Error: ${msg}`
+}
+
+const setResult = (original, reversed) => {
+    result.style.color = '#045502'
+    result.textContent = `${original} --> ${reversed}`
+}
+
+const reverseNum = () => {
+    if (Number.isNaN(input.value) || input.value === '') setError('Please input a valid number')
+    else {
+        const number = input.value
+        if (number.length !== 8) {
+            setError('Please input an 8-digit number')
+        } else {
+            const reversedNumber =
+                parseFloat(number.toString().split('').reverse().join('')) * Math.sign(number)
```

Comment:
you don't need to do `number.toString()` here, as inputs from `<input type="number"></input>` is always a string even though you explicitly set type to number.